### PR TITLE
Add join function

### PR DESCRIPTION
**Pull request by i4ki**
- add join
- add simple tests

### CODE REVIEW

**Review by katcipis — APPROVED** [review]
LGTM

File: str.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Join concatenates the elements of `lst` into a new string
+# with values separated by `sep`. `Lst` must contains only
+# string values.
+# This function is the bastard brother of builtin `split`.
+fn join(lst, sep) {
```
Comment (vitorarins) [inline str.sh]: `lst` has to be a list right? Maybe add this to the comment. If not, ignore this comment.

File: str_test.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			"plan9/from/outer/space"
+			(plan9 from outer space)
+			"/"
+		)
+		(
+			"writing tests in shellscript we are making history"
+			(writing tests in shellscript we are making history)
+			" "
+		)
+	)
```
Comment (vitorarins) [inline str_test.sh]: Maybe add a test case that tries to join something other than a list.

**Review by vitorarins — COMMENTED** [review]
Great PR. Kudos for the `all_test` script, looks like a great idea to use in more projects.

File: str_test.sh (hunk 2 above)
Comment (i4ki) [reply str_test.sh]: Will do. Thanks!

**Review by i4ki — COMMENTED** [review]
I'll setup CI on this project also.

File: str.sh (hunk 1 above)
Comment (i4ki) [reply str.sh]: Yes, `lst` must be a list. I can make it explictly in the docs.

File: str_test.sh (hunk 2 above)
Comment (i4ki) [reply str_test.sh]: After I thought about it, I don't think this kind of test makes sense anymore. :-)

The `join` function doesn't fail if properly called. If user don't pass a list in `lst` then the interpreter will catch the mistake (when trying to index a non-list variable). The language is dynamic and I don't think it's useful testing for programmers mistake.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+notifications:
+  webhooks:
+    urls:
+      - https://webhooks.gitter.im/e/52ad02845e880cdca2cf
+    on_success: change
+    on_failure: always
+    on_start: never
+  email:
+    - tiago.natel@neoway.com.br
+    - tiago.katcipis@neoway.com.br
```
Comment (i4ki) [inline .travis.yml:28]: Who else is interested in showing up here?

@vitorarins ?

Comment (vitorarins) [reply .travis.yml:28]: Sure, sign me up

File: str_test.sh (hunk 2 above)
Comment (vitorarins) [reply str_test.sh]: Well, I thought about testing exactly that the program would abort properly and not accept the parameter and return some crazy result. But I think you are right, this should be something for the interpreter (nash) to worry about and add as test case.

**Pull request merged**